plugins/modules/zos_mount.py

```diff
@@ -0,0 +1,995 @@
+            fullcmd = fullcmd + ' NOSECURITY'
+            parmtext = parmtext + '\n      NOSECURITY'
+        else:
+            fullcmd = fullcmd + ' SECURITY'
+            parmtext = parmtext + '\n      SECURITY'
+
+        if len(sysname) > 1:
+            fullcmd = "{0} SYSNAME\\({1}\\)".format(fullcmd, sysname)
+            parmtext = "{0}\n      SYSNAME({1})".format(parmtext, sysname)
+
+        if len(automove) > 1:
+            fullcmd = fullcmd + ' ' + automove
+            parmtext = parmtext + '\n      ' + automove
+            if(len(automove_list) > 1):
+                fullcmd = fullcmd + '(' + automove_list + ')'
+                parmtext = parmtext + '(' + automove_list + ')'
+        parmtext = parmtext + parmtail
+
+    if gonna_unmount:     # unmount/remount
+        fullumcmd = "tsocmd UNMOUNT FILESYSTEM\\(\\'{0}\\'\\)".format(src)
+        if len(unmount_opts) < 2:
+            unmount_opts = "NORMAL"
+            fullumcmd = fullcmd + ' ' + unmount_opts
+
+    comment = ''
+
+    if gonna_unmount:
+        if currently_mounted:
+            changed = True
+            if module.check_mode is False:
+                try:
+                    (rc, stdout, stderr) = module.run_command(
+                        fullumcmd, use_unsafe_shell=True)
+                    comment += "ran unmount command\n"
+                    currently_mounted = False
+                except Exception as err:
+                    module.fail_json(msg=str(err), stderr=str(res_args))
+            else:
+                comment += "(unmount) NO Action taken: ANSIBLE CHECK MODE\n"
+                stdout = 'ANSIBLE CHECK MODE'
+        else:
+            comment += "Unmount called on data set that is not mounted.\n"
+
+    if gonna_mount:
+        if currently_mounted is False:
+            changed = True
+            if module.check_mode is False:
+                try:
+                    (rc, stdout, stderr) = module.run_command(
+                        fullcmd, use_unsafe_shell=True)
+                    comment = "ran command\n"
+                except Exception as err:
+                    module.fail_json(msg=str(err), stderr=str(res_args))
+            else:
+                comment += "(mount) NO Action taken: ANSIBLE CHECK MODE\n"
+                stdout = 'ANSIBLE CHECK MODE'
+        else:
+            comment += "Mount called on data set that is already mounted.\n"
+
+    rc = 0
+    stdout = stderr = None
+
+    if write_fstab and module.check_mode is False:
+        fst_du = data_set.DataSetUtils(fstab)
+        fst_exists = fst_du.exists()
+        if fst_exists is False:
+            module.fail_json(
+                msg="fstab set member (" + fstab + ") doesn't exist",
+                stderr=str(res_args)
+            )
+
+        tmp_file = tempfile.NamedTemporaryFile(delete=True)
+        tmp_file_filename = tmp_file.name
+        tmp_file.close()
+
+        fullcmd = "cp \"//'" + fstab + "'\" " + tmp_file_filename
```

Comment:
You may be able to leverage the `copy` utility in module_utils to perform these copies

Reply:
looking into that

Reply:
I'm not sure about the copy module... it's looking for things like remote sftp port...wouldn't that be over-complicating something that (like a bpxprm file) really has to be local?

Reply:
Asif created a module utility named `copy` available in module_utils, which the `zos_copy` module and some others leverage to perform copies. The functions available in `copy` only operate on the z/OS target.

https://github.com/ansible-collections/ibm_zos_core/blob/dev/plugins/module_utils/copy.py

Reply:
For simple copying operations like these, another alternative would be to use ZOAU `Datasets.copy()` function.

Reply:
I'm trying the copy uss2ps and such...

Reply:
Thee copy utility, as it currently stands, messes with the file (I believe it's skipping the encoding).  It uses cp, but with a /F flag, which makes the contents hard to read.  I'm sticking with cp for now.